packages/shared/src/types/defaultTheme.ts

```diff
@@ -94,6 +94,11 @@ export interface Config {
    * @default false
    */
   enableScrollToTop?: boolean;
+  /**
+   * Whether to redirect to the closest locale when the user visits the site
+   * @default 'first-visit'
+   */
+  redirectToCloestLocale?: 'first-visit' | 'never';
```

Comment:
How about `localeRedirect?: 'auto' | 'never'`?

- auto: redirect by navigator.language
- never: do not redirect

And we can add more values in the future, such as redirect to the default language.

Reply:
`first-visit` explains current behavior, may be a `always` can be added in the future.

Reply:
During a redirect, the redirect behaviour can only be `first-visit`, whether redirecting to the browser's language or to the default language in the configuration.

For `always`, it will cause the user to be unable to switch languages normally. 

For example, if the default language of the document is English (which means there is no `/en` path on the URL), and the current language of the browser is Chinese, if the user wants to switch to English, then the logic of `always` will making it impossible for the user to switch to English.

Reply:
Makes sense, I'm open to adjustments.
I think there are at least two more strategies:
- Add a notification bar on the first visit, prompting the detection of a language that doesn't match the preference, and providing a link to confirm whether to redirect (so that users can choose not to redirect).
- Remember the current selection every time a language is chosen, and directly redirect to the user's selected language on the next visit.

Reply:
Agreed 😄, these two strategies will also have use cases. We can add new optional values in the future.

Reply:
Rename to `localeRedirect`